### PR TITLE
feat: Option to always document parameter types

### LINKED ISSUE #88: Option to write out types even when params are undocumented
Our library has more type hints than actual/manual documentation. It would be nice if we could put the types in the docs to give the user *something* even if our params are not documented. The way I would imagine this is that this extension adds a placeholder `:param foo:` if it's missing.

If you accept this feature request I would start implementing it.

### PR DESCRIPTION
**Pull request by untitaker**
Fix #88

### CODE REVIEW

**Comment by untitaker** [comment]
idk what was happening but this used to fail on 3.5

**Comment by untitaker** [comment]
@agronholm any update on this?

**Comment by agronholm** [comment]
I'm reviewing it now.

File: tests/roots/test-dummy/dummy_module.py [hunk 1]
```diff
@@ -177,3 +177,11 @@ def foo(
         :param x: foo
         """
         return 42
+
+
+def undocumented_function(x: int) -> str:
+    """
```
Comment (agronholm) [inline tests/roots/test-dummy/dummy_module.py]: Put this on a single line.

File: tests/test_sphinx_autodoc_typehints.py [hunk 2]
```diff
@@ -115,14 +115,16 @@ def test_process_docstring_slot_wrapper():
     assert not lines
 
 
+@pytest.mark.parametrize('always_document_param_types', [True, False])
 @pytest.mark.sphinx('text', testroot='dummy')
-def test_sphinx_output(app, status, warning):
+def test_sphinx_output(app, request, status, warning, always_document_param_types):
```
Comment (agronholm) [inline tests/test_sphinx_autodoc_typehints.py]: Why add the request fixture when you're not using it?

File: README.rst [hunk 3]
```diff
@@ -55,9 +55,13 @@ Options
 
 The following configuration options are accepted:
 
-* ``set_type_checking_flag`` (default: ``True``): if ``True``, set ``typing.TYPE_CHECKING`` to
+* ``set_type_checking_flag`` (default: ``False``): if ``True``, set ``typing.TYPE_CHECKING`` to
   ``True`` to enable "expensive" typing imports
 
+* ``always_document_param_types`` (default: ``False``): If ``False``, do not add type info for
```
[the remark this answers is not in the document]
Comment (untitaker) [reply README.rst]: Changed this but fwiw neither version creates more than one `<ul>`.

File: tests/test_sphinx_autodoc_typehints.py (hunk 2 above)
Comment (untitaker) [reply tests/test_sphinx_autodoc_typehints.py]: Yeah, this was a leftover from what I tried earlier

**Comment by untitaker** [comment]
@agronholm I believe I fixed everything

**Comment by untitaker** [comment]
Ah nvm I still need to debug the py 3.5 failure...

**Comment by untitaker** [comment]
@agronholm can you check again, the CI fixed itself

**Comment by agronholm** [comment]
Looks good now. Thanks!

**Pull request merged**